lib/src/eval/compiler/expression/prefix.dart

```diff
@@ -55,26 +59,30 @@ BuiltinValue _oneForType(TypeRef type, CompilerContext ctx) =>
         ? BuiltinValue(intval: 1)
         : BuiltinValue(doubleval: 1.0);
 
-Variable _handleDoubleOperands(PrefixExpression e, CompilerContext ctx) {
-  var V = compileExpression(e.operand, ctx);
-  final ref = compileExpressionAsReference(e.operand, ctx);
-  final L = ref.getValue(ctx);
+Variable _handleDoubleOperands(
+  PrefixExpression e,
+  CompilerContext ctx,
+  Reference V,
+  TypeRef type,
+  Variable L,
+) {
   var out = L;
 
   if (L.name != null) {
```

Comment:
What would happen if `L.name` was null? like if you did:
```dart
final list = [1]; 
final result = ++list[0];
print(result);
```

Reply:
I'm realizing that you just took that from the `postfix.dart` code so this one's my fault actually, sorry! Still, if you wouldn't mind that check should be removed from both prefix and postfix, it's incorrect

Reply:
Yeah, for `++list[0]`, this happens:

```plain
dart_eval runtime exception: type 'int' is not a subtype of type '$Value'
#0      BoxList.run (package:dart_eval/src/eval/runtime/ops/primitives.dart:255:69)
#1      Runtime.execute (package:dart_eval/src/eval/runtime/runtime.dart:787:12)
#2      Runtime.executeLib (package:dart_eval/src/eval/runtime/runtime.dart:775:12)
```